File: backend/api/schemas/skills.py
"""
api/schemas/skills.py
=====================
Pydantic response schemas for the Skill Intelligence router.
"""

from pydantic import BaseModel


class GrowingSkill(BaseModel):
    """A skill ranked by its YoY demand growth."""
    skill_name: str
    skill_category: str
    is_ai_related: bool
    growth_pct: float | None        # CANONICAL KPI
    avg_demand_score: float | None  # CANONICAL KPI


class AISkill(BaseModel):
    """An AI/ML-related skill with its demand metrics."""
    skill_name: str
    skill_category: str
    avg_demand_score: float | None
    avg_growth_pct: float | None
    total_mentions: int | None


class SkillCategorySummary(BaseModel):
    """Skill demand aggregated by category (Programming, Cloud, etc.)."""
    skill_category: str
    skill_count: int
    avg_growth_pct: float | None
    ai_skill_count: int


class SkillDemandTrend(BaseModel):
    """One point on a skill's demand trend chart (year over year)."""
    year: int
    skill_name: str
    avg_demand_score: float | None
    avg_growth_pct: float | None

File: frontend/src/types/skills.ts
// ─────────────────────────────────────────
// types/skills.ts
// Field names mirror backend/api/schemas/skills.py exactly.
// ─────────────────────────────────────────

// GET /api/v1/skills/top-growing
export interface GrowingSkill {
  skill_name: string
  skill_category: string
  is_ai_related: boolean
  growth_pct: number | null
  avg_demand_score: number | null
}

// GET /api/v1/skills/ai-skills
export interface AISkill {
  skill_name: string
  skill_category: string
  avg_demand_score: number | null
  avg_growth_pct: number | null
  total_mentions: number | null
}

// GET /api/v1/skills/by-category
export interface SkillCategorySummary {
  skill_category: string
  skill_count: number
  avg_growth_pct: number | null
  ai_skill_count: number
}

// GET /api/v1/skills/demand-trend?skill_name=Python
export interface SkillDemandTrend {
  year: number
  skill_name: string
  avg_demand_score: number | null
  avg_growth_pct: number | null
}

// Aggregate type for the page hook
export interface SkillsData {
  topGrowing: GrowingSkill[]
  aiSkills: AISkill[]
  byCategory: SkillCategorySummary[]
  demandTrend: SkillDemandTrend[]
}